pkg/sql/alter_database.go

```diff
@@ -2326,6 +2326,15 @@ func (n *alterDatabaseSetZoneConfigExtensionNode) startExec(params runParams) er
 		}
 	}
 
+	currentZone := zonepb.NewZoneConfig()
+	if currentZoneConfigWithRaw, err := params.p.Descriptors().GetZoneConfig(
+		params.ctx, params.p.Txn(), n.desc.ID,
+	); err != nil {
+		return err
+	} else if currentZoneConfigWithRaw != nil {
+		currentZone = currentZoneConfigWithRaw.ZoneConfigProto()
+	}
+
```

Comment:
I saw this snippet of code appearing multiple times. Do we want to make it a function (to avoid some code repetitions)?

Reply:
Since we are talking about only 5 lines of code, used in 4 places, my preference would be to keep the API simple. IMO the added maintenance overhead of making the API handle this very similar usage is not worth it. This article elaborates on some reasons why it can be better to add code duplication rather than introducing the wrong (or too many) API abstractions: https://sandimetz.com/blog/2016/1/20/the-wrong-abstraction?duplication